text/048-packaging-coq-native.md

```diff
@@ -0,0 +1,178 @@
+	* option `-native-compiler ondemand` (which becomes the default when compiling coq manually) preserves the previous default behavior (modulo the stdlib that is not precompiled anymore).
+* **item 3.** Optionally: to enhance `native_compute` support with old versions of Coq (the releases of Coq before 8.13 where **item 2** is implemented), update the `opam` packaging of the considered libraries.
+
+
+## Regarding item 1:
+
+* Add an opam metapackage `coq-native.1`:
+
+```
+opam-version: "2.0"
+maintainer: "Erik Martin-Dorel"
+authors: "Coq"
+homepage: "https://coq.inria.fr/"
+bug-reports: "https://github.com/coq/coq/issues"
+conflicts: [
+  "coq" {< "8.5"}
+]
+synopsis: "Package flag enabling coq's native-compiler flag"
+description: """
+This package acts as a package flag for the ambient switch, taken into
+account by coq (and possibly any coq library) to enable native_compute
+at configure time, triggering the installation of .coq-native/* files
+for the coq standard library and subsequent coq libraries.
+
+This implements item 1 of CEP #48 <https://github.com/coq/ceps/pull/48>.
+
+Remarks:
+
+1. you might face with issues installing this package flag under macOS,
+   see <https://github.com/coq/coq/issues/11178>.
+2. this package is not intended to be used as a dependency of other
+   packages (notably as installing or uninstalling this package may
+   trigger a rebuild of all coq packages in the ambient switch).
+3. the option set by this package will be automatically propagated to
+   coqc for coq >= 8.13 (but the packaging of coq libraries for
+   earlier versions of coq may need an update: for details, see item 3
+   of CEP #48 <https://github.com/coq/ceps/pull/48>).
+"""
+```
+
+* Add the following lines in the opam packages of `coq`:
+
+```diff
+diff --git a/packages/coq/coq.8.11.2/opam b/packages/coq/coq.8.11.2/opam
+index fdb19e9da7..3d3dee7570 100644
+--- a/packages/coq/coq.8.11.2/opam
++++ b/packages/coq/coq.8.11.2/opam
+@@ -23,6 +23,9 @@ depends: [
+   "num"
+   "conf-findutils" {build}
+ ]
++depopts: [
++  "coq-native"
++]
+ build: [
+   [
+     "./configure"
+@@ -33,7 +36,9 @@ build: [
+     "-libdir" "%{lib}%/coq"
+     "-datadir" "%{share}%/coq"
+     "-coqide" "no"
+-    "-native-compiler" {os = "macos"} "no" {os = "macos"}
++    "-native-compiler" "yes" {coq-native:installed} "no" {!coq-native:installed}
+   ]
+   [make "-j%{jobs}%"]
+   [make "-j%{jobs}%" "byte"]
+```
+
+## Regarding item 2:
+
+The `coqc` option `-native-compiler ondemand` corresponds to the default behavior since [4ad6855](https://github.com/coq/coq/commit/4ad6855504db2ce15a474bd646e19151aa8142e2) (8.5beta3).
+
+The following array, adapted from [this coq/coq#12564 comment](https://github.com/coq/coq/issues/12564#issuecomment-647464937), summarizes all behaviors (the changes are highlighted in bold):
```

Comment:
In https://github.com/coq/coq/issues/12564#issuecomment-647485222 there is an additional column, which could be interesting.